src/implementations/osmosis.rs

```diff
@@ -87,15 +85,16 @@ impl VaultToken for OsmosisDenom {
     fn query_total_supply(&self, deps: Deps) -> CwTokenResult<Uint128> {
         let amount_str = deps
             .querier
-            .query::<QuerySupplyOfResponse>(&QueryRequest::Stargate {
-                path: OsmosisTypeURLs::QuerySupplyOf.to_string(),
-                data: encode(QuerySupplyOfRequest {
-                    denom: self.to_string(),
-                }),
+            .query::<CoinMsg>(&QueryRequest::Stargate {
+                path: "/cosmos.bank.v1beta1.Supply".to_string(),
+                data: to_binary(&Supply {
+                    total: vec![CoinMsg {
+                        denom: self.to_string(),
+                        amount: "".to_string(),
+                    }],
```

Comment:
this doesnt look right, i thought the Supply query takes in `denom: String` as an argument. looking into this..